src/inmanta/data.py

```diff
@@ -459,7 +459,9 @@ class Project(BaseDocument):
 
     @gen.coroutine
     def delete_cascade(self):
-        yield Environment.delete_all(project=self.id)
+        environments = yield Environment.get_list(project=self.id)
+        for env in environments:
```

Comment:
Perhaps also add test to ensure nothing is deleted that shouldn't be (i.e. add another project/env and see that it is still there)